File: bubblesort/BubbleSort.java
public class BubbleSort {


    public static int[] bubbleSort(int[] array) {
        int next = 0;
        for (int i = 0; i < array.length; i ++) {
            int current = array[i];
            for (int j = i; j < array.length; j ++) {
                    if (j + 1 <= array.length -1) {
                        next = array[j + 1];
                        if (current > next) {
                            array[j + 1] = current;
                            array[i] = next;
                            current = next;
                        }
                    }
                }
            }
        return array;
    }



}

File: bubblesort/BubbleSort.py
def bubbleSort(array):
    isSorted = False
    while not isSorted:
        isSorted = True
        for i in range(len(array) - 1):
            if array[i] > array[i + 1]:
                swap(i, i + 1, array)
                isSorted = False

    return array

def swap(currently, next, array):
    array[currently], array[next] = array[next], array[currently]


if __name__ == '__main__':

    sorted = bubbleSort([5, 2, 4, 2, 1, 6, 9, 3, 7])

    for x in sorted:
        print(x)
